app/routes/circulars/circulars.server.ts

```diff
@@ -172,3 +173,13 @@ export async function put(subject: string, body: string, request: Request) {
     submitter: formatAuthor(user),
   })
 }
+
+export async function circularRedirect(query: string | undefined) {
+  if (!query) return
+  const validCircularSearchStyles = /(GCN\s*|Circular\s*|(^[0-9]+$))/i
```

Comment:
* [ ] This should match the entire string from beginning to end, so the regex should start with `^` and end with `$`.
* [ ] We don't care about leading or trailing whitespace, so add `\s*` between each token.
* [ ] Use an optional, [non-capturing group](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Groups_and_backreferences#using_groups) to make it so that the strings "gcn" and "circular" are fully optional. A non-capturing group has the syntax `(?:subpattern)`, and to make it optional, you can suffix it with a `?`, as in `(?:subpattern)?`.
* [ ] Do have a capturing group for the Circular number itself. Note that for silly historical reasons, there are a handful of Circulars with negative or half-integer IDs. `(-?\d+(?:\.\d)?)`